This patch displays the package license in 'opkg info' output.  An
'opkg update' operation is required because this information is not
stored in the local package database.

Index: trunk/libopkg/pkg.h
===================================================================
--- trunk.orig/libopkg/pkg.h
+++ trunk/libopkg/pkg.h
@@ -125,6 +125,7 @@ struct pkg
      char *maintainer;
      char *description;
      char *tags;
+     char *license;
      pkg_state_want_t state_want;
      pkg_state_flag_t state_flag;
      pkg_state_status_t state_status;
Index: trunk/libopkg/pkg_parse.c
===================================================================
--- trunk.orig/libopkg/pkg_parse.c
+++ trunk/libopkg/pkg_parse.c
@@ -267,6 +267,11 @@ pkg_parse_line(pkg_t *pkg, const char *l
 			parse_version(pkg, line);
 		break;
 
+	case 'L':
+		if((mask & PFM_LICENSE) && is_field("License", line))
+			pkg->license = parse_simple("License", line);
+		break;
+
 	case ' ':
 		if ((mask & PFM_DESCRIPTION) && reading_description) {
 			pkg->description = xrealloc(pkg->description,
Index: trunk/libopkg/pkg.c
===================================================================
--- trunk.orig/libopkg/pkg.c
+++ trunk/libopkg/pkg.c
@@ -97,6 +97,7 @@ int pkg_init(pkg_t *pkg)
      pkg->maintainer = NULL;
      pkg->section = NULL;
      pkg->description = NULL;
+     pkg->license = NULL;
      pkg->state_want = SW_UNKNOWN;
      pkg->state_flag = SF_OK;
      pkg->state_status = SS_NOT_INSTALLED;
@@ -613,6 +614,19 @@ void pkg_formatted_field(FILE *fp, pkg_t
                fprintf(fp, "Installed-Time: %lu\n", pkg->installed_time);
 	  }
 	  break;
+     case 'l':
+     case 'L': {
+	  /* License */
+	  if (strcasecmp(field, "License") == 0) {
+	       /* License */
+	       if (pkg->license) {
+                   fprintf(fp, "License: %s\n", pkg->license);
+	       }
+	  } else {
+	       goto UNKNOWN_FMT_FIELD;
+	  }
+     }
+	  break;
      case 'm':
      case 'M':
 	  if (strcasecmp(field, "Maintainer") == 0) {
@@ -761,6 +775,7 @@ void pkg_formatted_info(FILE *fp, pkg_t 
 	pkg_formatted_field(fp, pkg, "Conffiles");
 	pkg_formatted_field(fp, pkg, "Source");
 	pkg_formatted_field(fp, pkg, "Description");
+	pkg_formatted_field(fp, pkg, "License");
 	pkg_formatted_field(fp, pkg, "Installed-Time");
 	pkg_formatted_field(fp, pkg, "Tags");
 	fputs("\n", fp);
Index: trunk/libopkg/pkg_parse.h
===================================================================
--- trunk.orig/libopkg/pkg_parse.h
+++ trunk/libopkg/pkg_parse.h
@@ -53,6 +53,8 @@ int pkg_parse_from_stream_nomalloc(pkg_t
 #define PFM_TAGS		(1 << 25)
 #define PFM_VERSION		(1 << 26)
 
+#define PFM_LICENSE		(1u << 31)
+
 #define PFM_ALL	(~(uint)0)
 
 #endif
